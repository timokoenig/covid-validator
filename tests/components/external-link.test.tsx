import { Text } from '@chakra-ui/react'
import React from 'react'
import renderer from 'react-test-renderer'
import ExternalLink from '../../components/external-link'

it('renders correctly', () => {
  const tree = renderer
    .create(<ExternalLink href="https://covidvalidator.app">Link</ExternalLink>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with component', () => {
  const tree = renderer
    .create(
      <ExternalLink href="https://covidvalidator.app">
        <Text>Link</Text>
      </ExternalLink>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
