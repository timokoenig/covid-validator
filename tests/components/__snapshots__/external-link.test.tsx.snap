// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1x3w0s1{-webkit-margin-start:1px;margin-inline-start:1px;-webkit-margin-end:1px;margin-inline-end:1px;}",
      }
    }
    data-emotion="css 1x3w0s1"
  />,
  <a
    className="chakra-link css-1x3w0s1"
    href="https://covidvalidator.app"
    style={
      Object {
        "textDecoration": "underline",
      }
    }
    target="_blank"
  >
    Link
  </a>,
]
`;

exports[`renders correctly with component 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1x3w0s1{-webkit-margin-start:1px;margin-inline-start:1px;-webkit-margin-end:1px;margin-inline-end:1px;}",
      }
    }
    data-emotion="css 1x3w0s1"
  />,
  <a
    className="chakra-link css-1x3w0s1"
    href="https://covidvalidator.app"
    style={
      Object {
        "textDecoration": "underline",
      }
    }
    target="_blank"
  >
    <p
      className="chakra-text css-0"
    >
      Link
    </p>
  </a>,
]
`;
